### ISSUE TITLE
eachine e58 wifi fpv drone

### ISSUE THREAD

**Issue by hope-miky** [LukasMaly/wifi-ufo-drone#1]
Does this code work for eachine e58 wifi fpv drone?

**Comment by ujshriki** [comment]
I tested this code with echine e58 a few months ago.
The controls work ok, but I couldn't get any video stream.

**Comment by hope-miky** [comment]
But noting is working for me, I have connected to the wifi and run both the scripts as sudo but got an error saying could not find the device

**Comment by ujshriki** [comment]
First, make sure you connect your PC to the drone's WiFi.
your PC has IP 192.168.0.2 and drone is on 192.168.0.1
I used wireshark to figure out the E58 uses port 50000 instead of 40000 written in this code.
I also changed the iface to use the correct one.

scapy also made some trouble so I change the import:
(-) from scapy.all import * 
(+) from scapy import *
(+) from scapy.layers.inet import *

good luck.

**Comment by powderluv** [comment]
I did this to get it to work:

 UDP_HEARTBEAT_DATA = bytearray([0x63, 0x63, 0x01, 0x00, 0x00, 0x00, 0x00])
@@ -19,9 +20,9 @@ class DroneControl(object):
     def __init__(self):
         self._src = '192.168.0.2'
         self._dst = '192.168.0.1'
-        self._iface = 'en0'
+        self._iface = 'wlp0s20f3'
         self._tcp_dport = 7060
-        self._udp_dport = 40000
+        self._udp_dport = 50000
         self._udp_sport = random.randint(32768, 49152)


BTW has anyone got the video stream functional ?

**Comment by shouryashashank** [comment]
Does this work on the same protocol as the ky ufo app?  I have toy drone that I am able to control using that app. I wanted to know if there is something I can use to connect to the drone with my pc and run automated scrips on

**Comment by nitishmeswal** [comment]
> Does this work on the same protocol as the ky ufo app? I have toy drone that I am able to control using that app. I wanted to know if there is something I can use to connect to the drone with my pc and run automated scrips on

did it work for you?

**Comment by shouryashashank** [comment]
yes it connected I used some other port, I don't exactly remember, what I used. But the major issue is the input lag, it is almost impossible to control over wifi.

**Comment by shouryashashank** [comment]
> > Does this work on the same protocol as the ky ufo app? I have toy drone that I am able to control using that app. I wanted to know if there is something I can use to connect to the drone with my pc and run automated scrips on
> 
> did it work for you?

yes it connected I used some other port, I don't exactly remember, what I used. But the major issue is the input lag, it is almost impossible to control over wifi.

**Comment by nitishmeswal** [comment]
okay,can you share me the synopsis? maybe i can contribute
